saleor/payment/utils.py

```diff
@@ -69,24 +102,24 @@ def create_payment(**payment_data):
 
 
 def create_transaction(
-        payment: Payment,
-        token: str,
-        kind: str,
-        is_success: bool,
-        amount: Decimal,
-        currency: str,
-        gateway_response: Optional[Dict] = None) -> Transaction:
-    if not gateway_response:
+        payment: Payment, kind: str, payment_information: Dict,
+        gateway_response: Dict = None, error_msg=None) -> Transaction:
+    """Creates a Transaction based on transaction kind and gateway response."""
+    if gateway_response is None:
         gateway_response = {}
+
     txn, _ = Transaction.objects.get_or_create(
         payment=payment,
-        token=token,
-        kind=kind,
-        is_success=is_success,
-        amount=amount,
-        currency=currency,
+        kind=gateway_response.get('kind', kind),
+        token=gateway_response.get(
+            'transaction_id', payment_information['token']),
+        is_success=gateway_response.get('is_success', False),
+        amount=gateway_response.get('amount', payment_information['amount']),
```

Comment:
As amount and currency are required fields, I think there is no needs to provide the default values?

Reply:
Added a comment before this whole block 👍